### PR TITLE
koordlet: support collect BE CPU metric

### PR DESCRIPTION
**Pull request by jasonliu747**
Signed-off-by: Jason Liu <jasonliu747@gmail.com>

### Ⅰ. Describe what this PR does
This PR will make metricadvisor be able to collect BE CPU metric.

### CODE REVIEW

File: pkg/koordlet/metricsadvisor/be_collector.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+	return limit, nil
+}
+
+func (c *collector) getBECPURequestSum() resource.Quantity {
+	requestSum := int64(0)
+	for _, podMeta := range c.statesInformer.GetAllPods() {
+		pod := podMeta.Pod
+		if apiext.GetPodQoSClass(pod) == apiext.QoSBE {
+			for _, container := range pod.Spec.Containers {
+				containerCPUReq := util.GetContainerBEMilliCPURequest(&container)
```
Comment (saintube) [inline pkg/koordlet/metricsadvisor/be_collector.go:107]: Could we use `GetPodBEMilliCPURequest()` instead?  Resource requests of init containers and pod overhead should get supported in the future.

File: pkg/koordlet/metricsadvisor/collector.go [hunk 2]
```diff
@@ -63,7 +63,7 @@ type contextRecord struct {
 
 type collectContext struct {
 	// record latest cpu stat for calculate resource used
-	// lastBeCPUStat contextRecord
+	lastBeCPUStat        contextRecord
```
Comment (saintube) [inline pkg/koordlet/metricsadvisor/collector.go:66]: how about `lastBECPUStat`?

File: pkg/koordlet/metricsadvisor/be_collector.go (hunk 1 above)
Comment (jasonliu747) [reply pkg/koordlet/metricsadvisor/be_collector.go:107]: sure. btw, I saw you left a TODO in `GetPodBEMilliCPURequest()`, make sure you don't forget about it.

File: pkg/koordlet/metricsadvisor/collector.go (hunk 2 above)
Comment (jasonliu747) [reply pkg/koordlet/metricsadvisor/collector.go:66]: sure

File: pkg/koordlet/metricsadvisor/be_collector.go [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+	return limit, nil
+}
+
+func (c *collector) getBECPURequestSum() resource.Quantity {
+	requestSum := int64(0)
+	for _, podMeta := range c.statesInformer.GetAllPods() {
+		pod := podMeta.Pod
+		if apiext.GetPodQoSClass(pod) == apiext.QoSBE {
+			requestSum = util.GetPodBEMilliCPURequest(pod)
```
Comment (saintube) [inline pkg/koordlet/metricsadvisor/be_collector.go:106]: Here is a typo makes `requestSum +=` as `requestSum =`. And please note `util.GetPodBEMilliCPURequest()` can return `-1` when pod does not specify BatchCPU

**Review by saintube — CHANGES_REQUESTED** [review]

File: pkg/koordlet/metricsadvisor/be_collector.go (hunk 3 above)
Comment (jasonliu747) [reply pkg/koordlet/metricsadvisor/be_collector.go]: my bad, fixed.

File: pkg/koordlet/metricsadvisor/be_collector.go [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+func getBECPURealMilliLimit() (int, error) {
+	limit := 0
+
+	cpuSet, err := util.GetRootCgroupCurCPUSet(corev1.PodQOSBestEffort)
+	if err != nil {
+		return 0, err
+	}
+	limit = len(cpuSet) * 1000
+
+	cfsQuota, err := util.GetRootCgroupCurCFQuota(corev1.PodQOSBestEffort)
```
Comment (hormes) [inline pkg/koordlet/metricsadvisor/be_collector.go:77]: GetRootCgroupCurCF[S]Quota?

**Pull request merged**